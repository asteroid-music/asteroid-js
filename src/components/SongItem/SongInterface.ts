/**
 * Interface for information on a song.
 * Note that songs pulled from get requests to "/queue" also have a
 * 'votes' property whereas other songs may not.
 * Details for each individual property is given in comments below.
 */
interface SongInterface {
    /** Name of the song's artist */
    artist: string,

    /** Name of the song */
    song: string,

    /** Duration of the song, in seconds (float if necessary) */
    duration: number,

    /** Name of the album that the song comes from */
    album: string,

    /** Unique ID that the song is stored under in the backend database */
    _id: string,

    /**
     * Number of votes cast for the particular song.
     * Optional, as only given for songs in the current play queue.
     */
    votes?: number
}

export default SongInterface;
