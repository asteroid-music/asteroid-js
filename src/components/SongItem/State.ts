/**
 * State interface for the <SongItem> component
 * Details for each individual property is given in comments below.
 */
interface SongItemState {
    /** Whether to display song information on separate rows **/
    unfolded: boolean,

    /**
     * The change in vote for this song due to the user since this component
     * was created. I.e the value 0 if the user hasn't voted on this song, 1 if
     * they have upvoted this song, and -1 if they have downvoted this song.
     */
    voteState: number
}

export default SongItemState;
