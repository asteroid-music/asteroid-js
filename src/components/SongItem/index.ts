//Main component
export { AsteroidSongItem, SongInterface } from './SongItem';


