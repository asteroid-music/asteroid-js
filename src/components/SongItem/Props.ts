//Imports from local directory
import SongInterface from './SongInterface';

/**
 * Props interface for the <SongItem> component
 * Details for each individual property is given in comments below.
 */
interface SongItemProps {
    /** The song to display information for - see SongInterface **/
    song: SongInterface,

    /** Whether to show upvote and downvote buttons for the song **/
    voteButtons?: boolean,

    /** Whether to initially render the SongItem unfolded **/
    unfolded?: boolean
}

export default SongItemProps;
