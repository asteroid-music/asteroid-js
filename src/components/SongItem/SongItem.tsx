//Imports from external 'react' module
import React from 'react';
import type { ElementType } from 'react';

//Imports from external '@material-ui/core' module
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';

//Imports from external '@material-ui/core/colors' module
import green from '@material-ui/core/colors/green';
import red from '@material-ui/core/colors/red';

//Imports from external '@material-ui/icons' module
import ArrowUpwardIcon from '@material-ui/icons/ArrowUpward';
import ArrowDownwardIcon from '@material-ui/icons/ArrowDownward';
import UnfoldMore from '@material-ui/icons/UnfoldMore';
import UnfoldLess from '@material-ui/icons/UnfoldLess';
import IconButton from '@material-ui/core/IconButton';

//Import from external 'axios' module
import axios from 'utils/axios';

//Imports from local directory
import SongInterface from './SongInterface';
import SongItemProps from './Props';
import SongItemState from './State';

/**
 * <SongItem> component allowing interaction
 * with a single song item.
 */
class SongItem extends React.Component<SongItemProps,SongItemState> {
    constructor(props) {
        super(props);
        this.state = {
            unfolded: this.props.unfolded,
            voteState: 0
        };
    }

    unfoldedSecondaryNode() {
        let song = this.props.song;
        let dur = this._asTime(song.duration);
        return <div>
            <Typography>Artist: {song.artist}</Typography>
            <Typography>Album: {song.album}</Typography>
            <Typography>Duration: {dur}</Typography>
        </div>
    }

    foldedSecondaryText() {
        let song = this.props.song;
        return song.artist + " - " + song.album + " - " + this._asTime(song.duration);
    }

    _vote(value: number) {
        let voteState = this.state.voteState;
        axios.post<{message: string}>("/queue",{
            id: this.props.song?._id,
            vote: value
        }).catch(
            (error) => {
                this.setState({voteState:voteState});
            }
        );
    }

    _asTime(seconds: number) {
        let secs: number = seconds % 60;
        let mins: number = (seconds-secs)/60;
        return mins.toString()+":"+secs.toString();
    }

    upvoteCallback() {
        let voteState = this.state.voteState;
        if (voteState == 1) {
            this._vote(-1);
            this.setState({voteState:0});
        } else {
            this._vote(1-voteState);
            this.setState({voteState:1});
        }
    }

    downvoteCallback() {
        let voteState = this.state.voteState;
        if (voteState == -1) {
            this._vote(1);
            this.setState({voteState:0});
        } else {
            this._vote(voteState-1);
            this.setState({voteState:-1});
        }
    }

    render () {
        let song = this.props.song;

        let unfolded = this.state.unfolded;
        let voteState = this.state.voteState;

        let textSecondary = unfolded
            ? this.unfoldedSecondaryNode()
            : this.foldedSecondaryText();

        let foldButton = unfolded
            ? <IconButton onClick={() => {this.setState({unfolded:false});}}><UnfoldLess /></IconButton>
            : <IconButton onClick={() => {this.setState({unfolded:true});}}><UnfoldMore /></IconButton>

        let otherProps = unfolded
            ? {
                secondaryTypographyProps: {component:"div" as ElementType}
            }
            : {
                primaryTypographyProps: {noWrap:true},
                secondaryTypographyProps: {noWrap:true}
            };

        let upButtonStyle = (voteState === 1)
            ? {backgroundColor: green[100]}
            : {};

        let downButtonStyle = (voteState === -1)
            ? {backgroundColor: red[100]}
            : {};

        let upButton = this.props.voteButtons && <IconButton
            style={upButtonStyle}
            onClick={() => {this.upvoteCallback();}}
        >
            <ArrowUpwardIcon
                style={{ color: green[500] }}
            />
        </IconButton>

        let downButton = this.props.voteButtons && <IconButton
            style={downButtonStyle}
            onClick={() => {this.downvoteCallback();}}
        >
            <ArrowDownwardIcon
                style={{ color: red[500] }}
            />
        </IconButton>

        let votesInfo = song.votes && <Typography>{song.votes+voteState}</Typography>

        return <ListItem>
            <ListItemText
                primary={song.song}
                secondary={textSecondary}
                {...otherProps}
            />
            {upButton}
            {votesInfo}
            {downButton}
            {foldButton}
        </ListItem>
    }
}

export default SongItem;
