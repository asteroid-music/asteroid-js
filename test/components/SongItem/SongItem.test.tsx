import React from 'react';
import renderer from 'react-test-renderer';
import SongItem, {SongInterface} from '../../../src/components/SongItem';

//Test SongInterface without votes
const testSongVoteless: SongInterface = {
    artist: "This is a test artist name",
    song: "This is a test song name",
    duration: 556,
    album: "This is a test album name",
    id: 1,
}

//Test SongInterface with votes
const testSongVoteful: SongInterface = {
    votes: 3,
    ...testSongVoteless
}

//Test that the <SongItem> components render correctly
test.each([
    {song:testSongVoteless},
    {song:testSongVoteful},
    {song:testSongVoteless,unfolded:true},
    {song:testSongVoteful,unfolded:true},
    {song:testSongVoteless,voteButtons:true},
    {song:testSongVoteful,voteButtons:true},
    {song:testSongVoteless,unfolded:true,voteButtons:true},
    {song:testSongVoteful,unfolded:true,voteButtons:true}
])("<SongItem {...%o}> renders correctly", (props) => {
    const item = renderer
        .create(<SongItem {...props} />)
        .toJSON();
    expect(item).toMatchSnapshot();
});
