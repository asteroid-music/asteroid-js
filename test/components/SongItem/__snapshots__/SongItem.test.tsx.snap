// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SongItem /> basic rendering renders correctly with a voteful song 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
    >
      This is a test song name
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-displayBlock"
    >
      This is a test artist name - This is a test album name - 9:16
    </p>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    3
  </p>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.83L15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</li>
`;

exports[`<SongItem /> basic rendering renders correctly with a voteless song 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
    >
      This is a test song name
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-displayBlock"
    >
      This is a test artist name - This is a test album name - 9:16
    </p>
  </div>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.83L15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
        />
      </svg>
    </span>
  </button>
</li>
`;

exports[`<SongItem /> interaction: voting decreases the number of votes by one when the downvote button is hit and POST request is successful 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
    >
      This is a test song name
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-displayBlock"
    >
      This is a test artist name - This is a test album name - 9:16
    </p>
  </div>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "color": "#4caf50",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    2
  </p>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#ffcdd2",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "color": "#f44336",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.83L15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</li>
`;

exports[`<SongItem /> interaction: voting increases the number of votes by one when the upvote button is hit and POST request is successful 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
    >
      This is a test song name
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-displayBlock"
    >
      This is a test artist name - This is a test album name - 9:16
    </p>
  </div>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#c8e6c9",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "color": "#4caf50",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    4
  </p>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "color": "#f44336",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.83L15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</li>
`;

exports[`<SongItem /> prop: unfolded renders correctly when unfolded by default 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      This is a test song name
    </span>
    <div
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
    >
      <div>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Artist: 
          This is a test artist name
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Album: 
          This is a test album name
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Duration: 
          9:16
        </p>
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 18.59L8.83 20 12 16.83 15.17 20l1.41-1.41L12 14l-4.59 4.59zm9.18-13.18L15.17 4 12 7.17 8.83 4 7.41 5.41 12 10l4.59-4.59z"
        />
      </svg>
    </span>
  </button>
</li>
`;

exports[`<SongItem /> prop: voteButtons renders correctly with a voteful song and vote buttons 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
    >
      This is a test song name
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-displayBlock"
    >
      This is a test artist name - This is a test album name - 9:16
    </p>
  </div>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "color": "#4caf50",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    3
  </p>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "color": "#f44336",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.83L15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</li>
`;

exports[`<SongItem /> prop: voteButtons renders correctly with a voteless song and vote buttons 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
    >
      This is a test song name
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-displayBlock"
    >
      This is a test artist name - This is a test album name - 9:16
    </p>
  </div>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "color": "#4caf50",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
        />
      </svg>
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "color": "#f44336",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
        />
      </svg>
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.83L15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
        />
      </svg>
    </span>
  </button>
</li>
`;
